ctapipe_io_lst/__init__.py

```diff
@@ -184,51 +197,106 @@ def fill_lst_event_container_from_zfile(self, event):
         event_container.ped_id = event.ped_id
         event_container.module_status = event.lstcam.module_status
         event_container.extdevices_presence = event.lstcam.extdevices_presence
-        event_container.tib_data = event.lstcam.tib_data
-        event_container.cdts_data = event.lstcam.cdts_data
+
+        # unpack TIB data
+        rec_fmt = '=IHIBB'
+        rec_unpack = struct.Struct(rec_fmt).unpack_from
+        unpacked_tib = rec_unpack(event.lstcam.tib_data)
+        event_container.tib_event_counter = unpacked_tib[0]
+        event_container.tib_pps_counter = unpacked_tib[1]
+        event_container.tib_tenMHz_counter = unpacked_tib[2]
+        event_container.tib_stereo_pattern = unpacked_tib[3]
+        event_container.tib_masked_trigger = unpacked_tib[4]
+        #event_container.cdts_data = event.lstcam.cdts_data
         event_container.swat_data = event.lstcam.swat_data
-        event_container.counters = event.lstcam.counters
+
+        # unpack CDTS data
+        rec_fmt = '=IIIQQBBB'
+        rec_unpack = struct.Struct(rec_fmt).unpack_from
+        unpacked_cdts = rec_unpack(event.lstcam.cdts_data)
+        event_container.ucts_event_counter = unpacked_cdts[0]
+        event_container.ucts_pps_counter = unpacked_cdts[1]
+        event_container.ucts_clock_counter = unpacked_cdts[2]
+        event_container.ucts_timestamp = unpacked_cdts[3]
+        event_container.ucts_camera_timestamp = unpacked_cdts[4]
+        event_container.ucts_trigger_type = unpacked_cdts[5]
+        event_container.ucts_white_rabbit_status = unpacked_cdts[6]
+
+        # unpack Dragon counters
+        rec_fmt = '=HIIIQ'
+        rec_len = struct.calcsize(rec_fmt)
+        rec_unpack = struct.Struct(rec_fmt).unpack_from
+
+        event_container.pps_counter = np.zeros(self.camera_config.lstcam.num_modules)
+        event_container.tenMHz_counter = np.zeros(self.camera_config.lstcam.num_modules)
+        event_container.event_counter = np.zeros(self.camera_config.lstcam.num_modules)
+        event_container.trigger_counter = np.zeros(self.camera_config.lstcam.num_modules)
+        event_container.local_clock_counter = np.zeros(self.camera_config.lstcam.num_modules)
+        for mod in range(self.camera_config.lstcam.num_modules):
+
+            words=event.lstcam.counters[mod*rec_len:(mod+1)*rec_len]
```

Comment:
This looks like you could use `np.frombuffer` here to avoid this loop and fill the arrays directly

Reply:
Thank for the suggestion Max, unfortunately it is not immediate to me how to do, could you please show it to me?

Reply:
`counters` is just bytes, right? Bytes with a specific meaning. If i read the code correct, you have for each module, 

pps_counter, uint2
10 MHz couner,  uint4
event_counter uint4
trigger_counter uint4
local_clock_counter uint8

So yo load this immediatly into one array, you can do:
```
module_data = np.frombuffer(lstcam.counters, dtype=[('pps_counter', 'uint2'), ('counter_10Mhz', 'uint4'), ...])
```

Reply:
See https://docs.scipy.org/doc/numpy/reference/generated/numpy.frombuffer.html

Reply:
Thanks Max, I will take into account the possibility to change as you suggest. But is it mandatory now? We are trying to put things all together and I would not change the code related to these counters so in hurry. If the code as it is now is acceptable, please accept the PR, I promise I will make the change soon

Reply:
I  do not need to accept anything here, this is basically your repository. And right now, no
approved PR reviews are necessary to merge. 

I don't have the time and the in-depth knowledge of LST to properly review this code. 
There are to much LST internals involved.

 It is not ctapipe code anymore. You do not have to wait for approved  PRs. I was only making suggestions. I would however recommend, that those interested in LST reading, start reviewing here. So if there is anyone with enough knowledge of the LST internals, you should ask them if they could review.

Reply:
I understand, I thank you for your help, it is very welcome (only I am a bit in hurry with this code right now and I know that people are using it and it works even if it can be surely be improved)
Still two questions:
- Who has the right to accept the merge? 
- Do you understand why one of the travis check does not work?

Reply:
* You can just merge it yourself
* Yes, it is because the tagged version of ctapipe I tested against, requires pyhessio which could not be installed. I will add it on the current master.

Reply:
As long as there is no group of people reviewing here, you could also directly push to the master branchf for smaller, simple changes.